### PR TITLE
Handle content-type returned by the backend code

### PR DESCRIPTION
**Pull request by vdantu**
*Issue #, if available:*

*Description of changes:*

**To run CI tests on your changes refer [README.md](https://github.com/awslabs/mxnet-model-server/blob/master/ci/README.md)**

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: mms/context.py [hunk 1]
```diff
@@ -77,3 +80,6 @@ def report_status(self, code, reason_phrase=None):
 
     def add_response_property(self, key, value):
         self._response_header[key] = value
+
+    def get_response_property(self):
```
Comment (piyushghai) [inline mms/context.py]: Should this not be ```get_response_headers```  ?

File: mms/tests/unit_tests/test_worker_service.py [hunk 2]
```diff
@@ -40,7 +40,7 @@ def test_with_nil_request(self, service):
             service.retrieve_data_for_inference(None)
 
     def test_valid_req(self, service):
-        input_batch, req_to_id_map = service.retrieve_data_for_inference(self.data)
+        headers, input_batch, req_to_id_map = service.retrieve_data_for_inference(self.data)
```
Comment (piyushghai) [inline mms/tests/unit_tests/test_worker_service.py]: Just curious, is this the only unit test that is affected by this change ?

**Comment by piyushghai** [comment]
@vdantu Can you also rebase this PR with the latest master ?

File: mms/context.py (hunk 1 above)
Comment (vdantu) [reply mms/context.py]: The field being returned is called "response_property" and other APIs are named as such. So I wouldn't want to change the convention.

File: mms/tests/unit_tests/test_worker_service.py (hunk 2 above)
Comment (vdantu) [reply mms/tests/unit_tests/test_worker_service.py]: Yes, this is seems to be the only one affected. I think I will update this test though, to test for headers, even though this is an optional field.

File: mms/protocol/otf_message_handler.py [hunk 3]
```diff
@@ -67,8 +67,15 @@ def create_predict_response(ret, req_id_map, message, code):
         msg += struct.pack("!i", len(buf))
         msg += buf
 
-        # TODO: retrieve content_type from context
-        msg += struct.pack('!i', 0)  # content_type
+        if context is None:
+            msg += struct.pack('!i', 0)  # content_type
+        else:
+            response_headers = context.request_processor.get_response_property()
+            if response_headers.get('content-type') is None:
```
Comment (vdantu) [inline mms/protocol/otf_message_handler.py]: Add length check for content-type to avoid encoding empty strings.

File: mms/context.py [hunk 4]
```diff
@@ -77,3 +80,6 @@ def report_status(self, code, reason_phrase=None):
 
     def add_response_property(self, key, value):
```
Comment (frankfliu) [inline mms/context.py]: response header should be per request, we need update this API to support per request header

Comment (vdantu) [reply mms/context.py]: True, we need that.

File: mms/context.py (hunk 1 above)
Comment (vdantu) [reply mms/context.py]: Actually addressed this. This makes more sense as well.

File: mms/service.py [hunk 5]
```diff
@@ -60,20 +60,22 @@ def retrieve_data_for_inference(batch):
...
         for batch_idx, request_batch in enumerate(batch):
             req_id = request_batch.get('requestId').decode("utf-8")
             parameters = request_batch['parameters']
+            model_in_headers = dict()
 
             model_in = dict()
             for parameter in parameters:
                 model_in.update({parameter["name"]: parameter["value"]})
-
+                model_in_headers.update({parameter["name"]: {"content-type": parameter["contentType"]}})
```
Comment (piyushghai) [inline mms/service.py]: Small nit here : 
Should the parameter key be ```content-type``` and not ```contentType``` ?

Comment (vdantu) [reply mms/service.py]: `contentType` is what we get from a message from frontend. I think we can leave it as is. It is easy for us to differentiate between the message received from frontend and the actual metadata that we should use in multiple parts of the code.

**Pull request merged**